src/fundamental/macro-fundamental.h

```diff
@@ -564,3 +564,13 @@ static inline uint64_t ALIGN_OFFSET_U64(uint64_t l, uint64_t ali) {
 
 #define PTR_TO_SIZE(p) ((size_t) ((uintptr_t) (p)))
 #define SIZE_TO_PTR(u) ((void *) ((uintptr_t) (u)))
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#  define bswap_16(x) __builtin_bswap16(x)
+#  define bswap_32(x) __builtin_bswap32(x)
+#  define bswap_64(x) __builtin_bswap64(x)
+#else
+#  define bswap_16(x) (x)
+#  define bswap_32(x) (x)
+#  define bswap_64(x) (x)
+#endif
```

Comment:
i don't grok these? 

two issues: functions called "bswap" should always byte swap, regardles son which endianess we are running. stuff like beXtoh()/leXtoh() and friends otoh should do different stuff depending on endianess of the host cpu.

And also, glibc defines macros like this in byteswap.h. Userspace really should use those definitions. It makes sense to have similar definitions in EFI space, but that means they should be added to src/boot/efi/util.h because they are *only* necessary in EFI space.

NB: src/fundamental/ is the place for C code that shall be available in *both* userspace and efi space, and this hence does not qualify, as we only need it in efi space, userspace has it anyway, provided by libc